src/pages/Home/Home.js

```diff
@@ -1,24 +1,102 @@
 import { useHistory } from 'react-router-dom';
+import { useState } from 'react';
 
-function Home({ userToken, saveToken }) {
+function Home({ createList, joinList }) {
   let history = useHistory();
 
-  function handleClick(e) {
-    saveToken(e);
+  const [shareToken, setShareToken] = useState('');
+  const [joinListError, setJoinListError] = useState('');
 
-    if (userToken) {
-      history.push('/list');
-    }
+  const handleTokenChange = (event) => {
+    setJoinListError('');
+    setShareToken(event.target.value);
+  };
+
+  function handleCreateList() {
+    createList()
+      .then((success) => {
+        history.push('/list');
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  function handleJoinList(event) {
+    event.preventDefault();
+    setJoinListError(''); // remove message to ensure repeated error is read again by screen reader
+    joinList(shareToken)
+      .then((success) => {
+        history.push('/list');
+      })
+      .catch((err) => {
+        if (err.message === 'Invalid token')
+          setJoinListError(
+            'Sorry, that token is invalid. Please try again or create a new list.',
+          );
+        else
+          setJoinListError(
+            'Sorry, there was a problem connecting to the database. Please try again.',
+          );
+      });
   }
 
   return (
-    <main>
-      <h1>Welcome To Your Smart Shopping List</h1>
-      <button type="button" onClick={handleClick}>
-        Create a new list
-      </button>
-    </main>
+    <>
+      <header className="container__header header">
+        <h1 className="header__welcome">Welcome To Your Smart Shopping List</h1>
+      </header>
+
+      <main>
+        <div className="new-list">
+          <button
+            type="button"
+            className="new-list__button button"
+            onClick={handleCreateList}
+          >
+            Create a new list
+          </button>
+        </div>
+
+        <div className="container__separator">- or -</div>
+
+        <form
+          name="joinListForm"
+          onSubmit={handleJoinList}
+          className="join-list-form"
+        >
+          <p>Join an existing shopping list by entering a three word token.</p>
+          <div
+            aria-live="assertive"
+            className={`join-list-form__error error ${
+              joinListError ? 'error_on' : ''
+            }`}
+          >
+            {joinListError}
+          </div>
+          <label
+            className="join-list-form__label join-list-form__label_type_text label"
+            htmlFor="shareToken"
+          >
+            Share Token:
+          </label>
+          <input
+            className="join-list-form__text-field text-field"
+            type="text"
+            id="shareToken"
+            name="shareToken"
+            value={shareToken}
+            onChange={handleTokenChange}
+            maxLength="100"
+            required
+          />
```

Comment:
A couple things I think we may want to add for a11y reasons:

1. Add `aria-invalid="true"` to the input when appropriate to indicate when that form field is associated with an error
2. Since we will only have one field that can have errors, adding `aria-describedBy` on the input + and a matching id on the aria live region might be helpful to tie them together for a user focused on the field (though I want to test this more with the next suggestion to make sure it doesn't trigger too many screen reader announcements)
3. Adding a ref and focusing on the field when there is an error would be better UX, especially for disabled users

Reply:
Now that I'm really learning a11y, I have to agree with you here. I also wanted to thank you and Connie for teaching me about that. I  think the aria-invalid="true" would be useful. Overall I think you guys did a great job, and that the only suggestion I have is to maybe add a few more comments. If you do add it , let me know. I would definitely like to see it in action with the screen reader.

Reply:
After thinking about it and lots of testing different arrangements, I think it actually makes sense to have 2 "error regions" here: 

1. a hint for the input (the aria-describedby on the input refers to the id of this area), and
2. a summary of the new state of the entire form (aria-live or role=alert).

The first works in combination with aria-invalid=true when there is something wrong with the specific field (e.g. invalid token). This is really important, because it ensures any time the user focuses on the field, they will be told it has an error and read the describedby hint, which could be as simple as "Invalid token." 

The second I think would work better for our "connection problem" message, since it really has nothing to do with the specific field so would be misleading to connect it as describedby and make the field invalid. This area can also be used for a more detailed message for an invalid token, because it only gets triggered once (ideally right after form submission if the screen reader is being predictable). So something like "Sorry, your token is invalid. Please try again or create a new list" could go here. The images below should give you a better idea of what I mean. (If it seems too redundant to show both, we could always visually hide one message when we add final CSS.) 

![image](https://user-images.githubusercontent.com/67769490/127731639-3abc0fc1-cbd4-4d34-91bd-605fc7dba92d.png)

![image](https://user-images.githubusercontent.com/67769490/127731674-cc23451b-b5a2-4678-91aa-273e8ba75482.png)